ols/user_data_collection/data_collector.py

```diff
@@ -342,6 +342,9 @@ def gather_ols_user_data(data_path: str) -> None:
                 logger.error(
                     f"{e.__class__.__name__} - upload and data removal canceled"
                 )
+
+            # close the tarball to release mem
+            tarball.close()
```

Comment:
this seems like a good change but i'm not convinced this was the cause of the memory utilization.  did you do some tests that confirmed it?

the collector only runs every 2 hours (the first run is something like 5 mins after startup).  So it's hard to see how this could cause a rapid runaway of memory usage in the first few minutes after startup.

also the archives themselves are pretty small, so even if we were leaking the size of the tarfile, that would not explain hundreds of megs of usage, i don't think.

Reply:
(also i'd sort of expect python to close the tarball when the variable goes out of scope, but that may not be a valid assumption)